web/pandas_web.py

```diff
@@ -223,7 +224,23 @@ def home_add_releases(context):
         with open(pathlib.Path(context["target_path"]) / "releases.json", "w") as f:
             json.dump(releases, f, default=datetime.datetime.isoformat)
 
-        for release in releases:
+        # Map to create a list of (version, realse) pairs from releases
+        parsed_releases = map(lambda r: (version.parse(r["tag_name"]), r), releases)
+
+        # Sort the releases in descending order
+        sorted_releases = sorted(parsed_releases, reverse=True)
+
+        # Gathers minor versions that are NOT obsolete
+        latest_releases = []
+        prev_version = version.Version("0.0.0")
+        for v, release in sorted_releases:
+            if (v.major, v.minor) == (prev_version.major, prev_version.minor):
+                # This release is of obsolete version, so skip
+                continue
+            latest_releases.append(release)
```

Comment:
I think you can use `v` here instead of release. Since `str(v) == release` I think the render on the website should stay the same. And you don't need to make `releases` and `sorted_releases` lists (or generators) of tuples, but only of `Version` objects, which makes things easier.

Reply:
I don't think `str(v) == release` is true judging from the code below. I think there are fields like `assets` and `tag_name` in each `release`.

Reply:
@datapythonista Hi, can you give some more clarifications on this?

Reply:
Sorry for the delay. So, here you keep a tuple to deal with versions: `(Version(1, 5, 2), '1.5.2')`. You use the first element to compare versions, and the second to display it in the template. But the template will parse to string whatever you give it when creating the html. And this is true:

```python
>>> from packaging import version
>>> v152 = version.Version('1.5.2')
>>> str(v152)
'1.5.2'
```
The second element in the tuple is not needed, since you can give the first one to the template, and when rendered it will become the second. So, you can just use `Version(...)` instead of the tuple, and simplify the code, which is a bit tricky to follow.

Reply:
Hi, I am still a little confused. The first element in the tuple is indeed the object`Version(...)`, but it is parsed from `release[tag_name]` but not from `release` directly. Since each `release` is a dictionary with many fields including `assets`, `tag_name`, `published_at` etc, I can only think of making them key-values pairs in the list `(v, release)` and sort the list  by the key values `v`. In other word, the list `latest_releases` consists of tuple like `(Version(1,5,2), {...})` instead of `(Version(1, 5, 2), '1.5.2')`. So, I'm sorting the dictionaries by the versions and then feed the dictionaries to the next loop.